apps/rush-lib/src/start.ts

```diff
@@ -1,6 +1,25 @@
 // Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
 // See LICENSE in the project root for license information.
 
+import * as colors from 'colors';
+import * as semver from 'semver';
+
+// check to ensure that we are using a supported version of NodeJS, otherwise write a warning
+if (semver.satisfies(process.versions.node, '< 6.0.0')) {
+  console.error(colors.red(`You are using an outdated version of Node ("${process.versions.node}").`
+    + ` You should upgrade to Node 6 or Node 8.`));
+  process.exit(1);
+} else if (semver.satisfies(process.versions.node, '<= 6.4.0')) {
+  console.warn(colors.yellow(`You are using an outdated version of Node ("${process.versions.node}").`
+    + ` You should upgrade to Node >=6.5.0 or Node 8, as some Rush features may not work.`));
```

Comment:
>You should upgrade to Node >=6.5.0 or Node 8, as some Rush features may not work. [](start = 8, length = 81)

If we objectively believe this version of Node is likely to be missing APIs that we use, we shouldn't let people run Rush.